### PR TITLE
defines QNX implementation for rcutils_get_platform_library_name

### LINKED ISSUE #286: failure to record due to QNX not defined in rcutils_get_platform_library_name()
in https://github.com/ros2/rcutils/blob/a287e74c700deb5816946349fa35aae31b1476f6/src/shared_library.c#L183-L219
QNX is not defined so rcutils_snprintf() never gets called and variable "written" remains zero. This triggers the error checking code at https://github.com/ros2/rcutils/blob/a287e74c700deb5816946349fa35aae31b1476f6/src/shared_library.c#L220-L225 which causes a call to "ros2 bag record /topic" to fail

### PR DESCRIPTION
**Pull request by asobhy-qnx**
Fixes #286

### CODE REVIEW

**Review by ahcorde — APPROVED** [review]

File: src/shared_library.c [hunk 1]
```diff
@@ -180,7 +180,7 @@ rcutils_get_platform_library_name(
 
   int written = 0;
 
-#ifdef __linux__
+#if defined(__linux__) || defined(__QNXNTO__)
```
Comment (ahcorde) [inline src/shared_library.c:183]:
```suggestion
#if defined(__linux__) || defined(__QNXNTO__)
```

**Review by ahcorde — CHANGES_REQUESTED** [review]
Please check the linters (uncrustify)

**Review by ahcorde — APPROVED** [review]

**Comment by ahcorde** [comment]
Testing up-to rcutils

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12141/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7131/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/9884/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12040/)

**Comment by asobhy-qnx** [comment]
@dirk-thomas can this be back ported to foxy and dashing

**Comment by jacobperron** [comment]
Foxy backport is here: https://github.com/ros2/rcutils/pull/309

I don't believe this change is applicable to Dashing since there is no `shared_library.c`.

**Pull request merged**